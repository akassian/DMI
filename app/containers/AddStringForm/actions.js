/*
 * AddStringForm Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import { ADD_STRING, ADD_STRING_ERROR, CHANGE_STRING } from './constants';

/**
 * Changes the input field of the form
 *
 * @param  {string} string The new text of the input field
 *
 * @return {object} An action object with a type of CHANGE_STRING
 */
export function changeString(string) {
  return {
    type: CHANGE_STRING,
    string,
  };
}

/**
 * Add the string, this action starts the request saga
 *
 * @return {object} An action object with a type of ADD_STRING
 */
export function addString() {
  console.log('inside addString');
  return {
    type: ADD_STRING,
  };
}

/**
 * Dispatched when strings are added to backend
 *
 * @param  {array} strings The strings array from backend
 *
 * @return {object}       An action object with a type of FETCH_STRINGS passing the strings array
 */
// export function stringsFetched(strings) {
//   console.log('inside stringsFetched');
//   return {
//     type: FETCH_STRINGS_SUCCESS,
//     strings,
//   };
// }

/**
 * Dispatched when adding string to backend fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of ADD_STRING_ERROR passing the error
 */
export function stringAddError(error) {
  return {
    type: ADD_STRING_ERROR,
    error,
  };
}
