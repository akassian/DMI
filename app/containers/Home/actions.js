import {
  FETCH_STRINGS,
  FETCH_STRINGS_SUCCESS,
  FETCH_STRINGS_ERROR,
} from './constants';

/**
 * Load the strings, THIS ACTION STARTS THE REQUEST SAGA
 *
 * @return {object} An action object with a type of FETCH_STRINGS
 */
export function fetchStrings() {
  return {
    type: FETCH_STRINGS,
  };
}

/**
 * Dispatched when strings are fetched from backend - DISPATCHED BY SAGA
 *
 * @param  {array} strings The strings array from backend
 *
 * @return {object}        An action object with a type of FETCH_STRINGS passing the strings array
 */
export function stringsFetched(strings) {
  return {
    type: FETCH_STRINGS_SUCCESS,
    strings,
  };
}

/**
 * Dispatched when fetching strings from backend fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of FETCH_STRINGS_ERROR, passing the error
 */
export function stringsFetchError(error) {
  return {
    type: FETCH_STRINGS_ERROR,
    error,
  };
}
