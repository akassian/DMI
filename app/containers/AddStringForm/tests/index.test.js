/**
 * Test the AddStringForm container
 */

import React from 'react';
import { render } from 'react-testing-library';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import { AddStringForm, mapDispatchToProps } from '../index';
import { changeString, addString } from '../actions';
import configureStore from '../../../configureStore';

describe('<AddStringForm />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <AddStringForm
          loading={false}
          error={false}
          string=""
          onChangeString={jest.fn()}
          onSubmitString={jest.fn()}
          dispatchClearString={jest.fn()}
        />
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    describe('onChangeString', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onChangeString).toBeDefined();
      });

      it('should dispatch changeString when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        const text = 'test string';
        result.onChangeString({ target: { value: text } });
        expect(dispatch).toHaveBeenCalledWith(changeString(text));
      });
    });

    describe('onSubmitString', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onSubmitString).toBeDefined();
      });

      it('should dispatch addString when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.onSubmitString();
        expect(dispatch).toHaveBeenCalledWith(addString());
      });

      it('should preventDefault if called with event', () => {
        const preventDefault = jest.fn();
        const result = mapDispatchToProps(() => {});
        const evt = { preventDefault };
        result.onSubmitString(evt);
        expect(preventDefault).toHaveBeenCalledWith();
      });
    });
  });
});
