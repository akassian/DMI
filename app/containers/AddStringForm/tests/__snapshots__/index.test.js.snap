// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddStringForm /> should render and match the snapshot 1`] = `
<form
  class="Form-x91grh-0 kGLYhq"
>
  <div>
    <label
      for="string"
    >
      Text:
    </label>
  </div>
  <textarea
    cols="50"
    id="string"
    required=""
    rows="4"
  />
  <div>
    <input
      type="submit"
    />
    <button
      type="button"
    >
      Cancel
    </button>
  </div>
</form>
`;
